src/utils/listenerHelpers.ts

```diff
@@ -87,3 +88,79 @@ export function propGetSet<Obj extends object, Key extends keyof Obj>(
 ) {
   return [() => obj[key], (v: Obj[Key]) => (obj[key] = v)] as const;
 }
+
+// override all keyboard inputs entering a MathQuill field
+// and optionally prevent them from doing their default behavior
+export function hookIntoOverrideKeystroke(
+  // field for which to override kb inputs
+  mq: MathQuillField,
+  // callback function
+  // return false to override all other events
+  fn: (key: string, evt: KeyboardEvent) => boolean | undefined,
+  // higher priority --> runs first
+  priority: number,
+  // unique key to prevent adding duplicate hooks
+  key: string
+) {
+  // @ts-expect-error adding properties to functions
+  if (!mq.__options.overrideKeystroke.isMonkeypatchedIn) {
+    const unsub = attach(
+      ...propGetSet(mq.__options, "overrideKeystroke"),
+      (key, evt) => {
+        // @ts-expect-error adding properties to functions
+        for (const h of mq.__options.overrideKeystroke.handlers as {
+          fn: (key: string, evt: KeyboardEvent) => boolean | undefined;
+          priority: number;
+        }[]) {
+          if (h.fn(key, evt) === false) return [false, undefined];
+        }
+      }
+    );
+
+    // @ts-expect-error adding properties to functions
+    mq.__options.overrideKeystroke.unsub = unsub;
+    // @ts-expect-error adding properties to functions
+    mq.__options.overrideKeystroke.isMonkeypatchedIn = true;
+    // @ts-expect-error adding properties to functions
+    mq.__options.overrideKeystroke.handlers = [];
+  }
+
+  const handler = { priority, fn, key };
+
+  const overrideKeystroke = mq.__options
+    .overrideKeystroke as typeof mq.__options.overrideKeystroke & {
```

Comment:
This type assertion might not be needed if you change the type of `overrideKeystroke` in `MathQuillFieldOptions`